src/Kiota.Builder/Writers/Shell/ShellCodeMethodWriter.cs

```diff
@@ -125,19 +125,19 @@ private void WriteExecutableCommand(CodeMethod codeElement, CodeClass parentClas
             {
                 // Add output filter param
                 parameters.Add((OutputFilterParamType, OutputFilterParamName, null));
-                availableOptions.Add($"{InvocationContextParamName}.BindingContext.GetRequiredService<{OutputFilterParamType}>()");
+                availableOptions.Add($"{InvocationContextParamName}.BindingContext.GetService(typeof({OutputFilterParamType})) as {OutputFilterParamType} ?? throw new ArgumentNullException(\"{OutputFilterParamName}\")");

```

Comment:
what's the benefit here of switching to the non generic method and having to use null coalesce + exception?

Reply:
So I can remove the need for the dependency injection library

Reply:
meaning the generated code depends on few types?

Reply:
Yes. The `GetRequiredService` is an extension method in the Microsoft.Extensions.DependencyInjection library

Reply:
thanks for the additional context here.